### PR TITLE
[cling] Fix TLS in the cling JIT

### PR DESCRIPTION
**Pull request by Teemperor**
TLS is currently not suppored in the JIT. However, it's possible to
enable emulated TLS support in LLVM which means that we now support
TLS across all architectures. The performance downsides of this
should be neglectiable and can be easily worked around (by merging
TLS variables into a single one).

Patch created with a lot of help from Lang Hames and Pavel Labath!

### CODE REVIEW

**Comment by vgvassilev** [comment]
@phsft-bot build with flags -Dclingtest=On

**Comment by dpiparo** [comment]
Hi, merging this will unblock the TFuture/Async building blocks for parallelism. Is there any blocker preventing us from proceeding with the merge?

**Comment by vgvassilev** [comment]
I'd like to put that in after PR #1218. I expect that would happen by the end of the week.

**Comment by vgvassilev** [comment]
@phsft-bot build!

@Teemperor, could you fix formatting?

**Comment by vgvassilev** [comment]
@phsft-bot build!

File: interpreter/cling/lib/Interpreter/IncrementalExecutor.cpp [hunk 1]
```diff
@@ -72,11 +72,11 @@ CreateHostTargetMachine(const clang::CompilerInstance& CI) {
   std::string MCPU;
   std::string FeaturesStr;
 
-  return std::unique_ptr<TargetMachine>(TheTarget->createTargetMachine(Triple,
-                                        MCPU, FeaturesStr,
-                                        llvm::TargetOptions(),
-                                        Optional<Reloc::Model>(), CMModel,
-                                        OptLevel));
+  auto TM = std::unique_ptr<TargetMachine>(TheTarget->createTargetMachine(
```
Comment (Axel-Naumann) [inline interpreter/cling/lib/Interpreter/IncrementalExecutor.cpp]: Could you please split formatting and functional change into two commits? I'm terrible at filtering white-space parts of diffs from functional ones... Thanks, much appreciated!

Comment (Teemperor) [reply interpreter/cling/lib/Interpreter/IncrementalExecutor.cpp]: Alright, done.

**Comment by vgvassilev** [comment]
@phsft-bot build only on windows10/vc15

**Comment by vgvassilev** [comment]
@phsft-bot build just on windows10/vc15

**Pull request merged**